auth/store.go

```diff
@@ -110,23 +116,27 @@ type AuthStore interface {
 	// RoleList gets a list of all roles
 	RoleList(r *pb.AuthRoleListRequest) (*pb.AuthRoleListResponse, error)
 
-	// UsernameFromToken gets a username from the given Token
-	UsernameFromToken(token string) (string, bool)
+	// AuthInfoFromToken gets a username from the given Token and current revision number
+	// (The revision number is used for preventing the TOCTOU problem)
+	AuthInfoFromToken(token string) (string, uint64, bool)
```

Comment:
the user and rev are used together everywhere, it would probably be cleaner to have `type AuthUser struct { user string; rev int64}` instead of having them separate